Build failure: python311Packages.bambi

### Steps To Reproduce

Steps to reproduce the behavior:
1. build `python311Packages.bambi`

### Build log

```
=========================== short test summary info ============================
FAILED tests/test_built_models.py::test_ordinal_families[sratio-logit] - TypeError: 'NoneType' object cannot be interpreted as an integer
FAILED tests/test_built_models.py::test_ordinal_families[sratio-probit] - TypeError: 'NoneType' object cannot be interpreted as an integer
FAILED tests/test_built_models.py::test_ordinal_families[sratio-cloglog] - TypeError: 'NoneType' object cannot be interpreted as an integer
FAILED tests/test_model_construction.py::test_categorical_term - ValueError: Size length is incompatible with batched dimensions of paramete...
FAILED tests/test_model_construction.py::test_1d_group_specific - ValueError: Size length is incompatible with batched dimensions of paramete...
= 5 failed, 145 passed, 2 skipped, 99 deselected, 203 warnings in 1390.07s (0:23:10) =
```

### Additional context

This has already been reported upstream: https://github.com/bambinos/bambi/issues/805

-> The goal of this issue is to track this failure but also to ask whether we should skip the failing tests or mark this package as broken.
Leaving it as is wastes maintainers time (attempting to build it for nothing) and pollutes `nixpkgs-review` logs with _irrelevant_ failures.

### Notify maintainers

cc @bcdarwin @ferrine 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.30, NixOS, 24.05 (Uakari), 24.05.20240505.25865a4`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.2`
 - nixpkgs: `/nix/store/3pif36ks3f56py4wb1dkq6sh0nkf3ygj-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/bambi/default.nix
+++ b/pkgs/development/python-modules/bambi/default.nix
@@ -1,36 +1,36 @@
 {
   lib,
-  buildPythonPackage,
-  pythonOlder,
-  fetchFromGitHub,
-  setuptools,
   arviz,
+  blackjax,
+  buildPythonPackage,
+  fetchFromGitHub,
   formulae,
   graphviz,
+  numpyro,
   pandas,
   pymc,
-  blackjax,
-  numpyro,
   pytestCheckHook,
+  pythonOlder,
+  setuptools-scm,
 }:
 
 buildPythonPackage rec {
   pname = "bambi";
-  version = "0.13.0";
+  version = "0.14.0";
   pyproject = true;
 
-  disabled = pythonOlder "3.8";
+  disabled = pythonOlder "3.10";
 
   src = fetchFromGitHub {
     owner = "bambinos";
     repo = "bambi";
     rev = "refs/tags/${version}";
-    hash = "sha256-9+uTyV3mQlHOKAjXohwkhTzNe/+I5XR/LuH1ZYvhc8I=";
+    hash = "sha256-kxrNNbZfC96/XHb1I7aUHYZdFJvGR80ZI8ell/0FQXc=";
   };
 
-  nativeBuildInputs = [ setuptools ];
+  build-system = [ setuptools-scm ];
 
-  propagatedBuildInputs = [
+  dependencies = [
     arviz
     formulae
     graphviz
@@ -38,15 +38,20 @@ buildPythonPackage rec {
     pymc
   ];
 
-  preCheck = ''
-    export HOME=$(mktemp -d)
-  '';
+  # bayeux-ml is not available in nixpkgs
+  # optional-dependencies = {
+  #   jax = [ bayeux-ml ];
+  # };
 
   nativeCheckInputs = [
     blackjax
     numpyro
     pytestCheckHook
   ];
+
+  preCheck = ''
+    export HOME=$(mktemp -d)
+  '';
 
   disabledTests = [
     # Tests require network access
@@ -65,29 +70,38 @@ buildPythonPackage rec {
     "test_group_effects"
     "test_hdi_prob"
     "test_legend"
+    "test_model_with_group_specific_effects"
+    "test_model_with_intercept"
+    "test_model_without_intercept"
     "test_non_distributional_model"
     "test_normal_with_splines"
-    "test_predict_offset"
-    "test_predict_new_groups"
     "test_predict_new_groups_fail"
+    "test_predict_new_groups"
+    "test_predict_offset"
     "test_set_alias_warnings"
     "test_subplot_kwargs"
     "test_transforms"
     "test_use_hdi"
-    "test_with_groups"
     "test_with_group_and_panel"
+    "test_with_groups"
     "test_with_user_values"
+  ];
+
+  disabledTestPaths = [
+    # bayeux-ml is not available
+    "tests/test_alternative_samplers.py"
+    # Tests require network access
+    "tests/test_interpret.py"
+    "tests/test_interpret_messages.py"
   ];
 
   pythonImportsCheck = [ "bambi" ];
 
   meta = with lib; {
-    homepage = "https://bambinos.github.io/bambi";
     description = "High-level Bayesian model-building interface";
+    homepage = "https://bambinos.github.io/bambi";
     changelog = "https://github.com/bambinos/bambi/releases/tag/${version}";
     license = licenses.mit;
     maintainers = with maintainers; [ bcdarwin ];
-    # https://github.com/NixOS/nixpkgs/issues/310940
-    broken = true;
   };
 }

